tidy3d/components/data/monitor_data.py

```diff
@@ -615,24 +622,29 @@ def _interpolated_tangential_fields(self, coords: ArrayFloat2D) -> Dict[str, Dat
         return fields
 
     def outer_dot(
```

Comment:
if `field_data` is a `ModeData` then it's field components can now be `None`. do we explicitly make sure this is not the case anywhere when implementing these `dot` functions? (or other functions that use the fields?)

Reply:
I *think* given that `field_components` still returns `return {field_name: field for field_name, field in fields.items() if field is not None}`, there should be no change in behavior

Reply:
yea, I'm more worried about if the dot functions try to grab one of those components but it is missing, whether that is handled gracefully or you just get an internal KeyError

Reply:
Yeah that's always been a possibility in `FieldData`, so it's been handled, you should just hit this: https://github.com/flexcompute/tidy3d/blob/1f22335e92d92cf875a1f70e4832acd9515a4c8b/tidy3d/components/data/monitor_data.py#L394C1-L395

Reply:
That's right. All user-facing functions should handle missing components gracefully. Internal methods *do not* have those checks, for example: `ModeData._isel`, `ModeData._find_ordering_one_freq`, etc., but those are use-at-your-own-risk methods, so I guess the user should be responsible for the check.